### PR TITLE
Optimization of the Editor UI and UX

### PR DESCRIPTION
**Pull request by elzadj**
# Description

Optimizations for the Editor, including FSE. Also changed the loading animation (and it shows per a block now).

**IMPORTANT:** 
Please do ```npm ci --legacy-peer-deps``` after switching to the branch.
And CTRL+F5 on the editor to clear cache for the loader css.

# How Has This Been Tested?

Test everything you can. That's one of these PRs, yep.

# Test checklist

**_ Front/Back Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Test FSE
-   [ ] Test with pages from a previous Maxi version
-   [ ] Test with cloud patterns

**_ Pre-Code Testing _**

-   [ ] Test the component in normal state in sidebar
-   [ ] Test the component in hover state in sidebar (if hover exists)
-   [ ] Check that the settings work on frontend
-   [ ] Check that backend works and saves the settings after the editor reload
-   [ ] Same 1-4 points for the toolbar
-   [ ] Same 1-4 points on responsive
-   [ ] Test the block inside the grid (Container + Row + Column)
-   [ ] Test the block as a standalone block
-   [ ] Duplicate the block, test that the settings of the first do not affect the second
-   [ ] Test with 2 blocks of the same type
-   [ ] Test FSE
-   [ ] Test with pages from a previous Maxi version
-   [ ] Test with cloud patterns
-   [ ] Check no commented code and no unnecessary imports
-   [ ] Standards of the project have been followed
-   [ ] No errors/warnings on console

# Checklist

-   [ ] My code follows the style guidelines of this project
-   [ ] I have performed a self-review of my code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] My changes generate no new warnings/errors
-   [ ] I have added/updated tests that prove my fix is effective or that my feature works
-   [ ] New and existing unit tests pass locally with my changes

### Summary by CodeRabbit

- Test: Enhanced the reliability of end-to-end tests by introducing delays and improving keyboard interactions.
- Refactor: Optimized the application performance by implementing dynamic imports and lazy loading of components using the `@loadable/component` library.
- Bug Fix: Fixed a missing return statement in the `maxi_blocks_pro()` function, ensuring the function returns the expected content.
- Style: Simplified the `z-index` value of the `.maxi-components-popover` class by replacing a function call with a hardcoded value.
- Refactor: Improved the `AdvancedNumberControl` component by introducing a debounced change handler and prioritizing values for the range control.
- Test: Improved the reliability of tests by adding delays before typing input values and waiting for changes to take effect before making assertions.
- Refactor: Updated the `SuspendedBlock` and `withMaxiLoader` components to improve block rendering and consolidation.

### CODE REVIEW

**Review by Olekrut — CHANGES_REQUESTED** [review]
checked with this page 
[Data Company.txt](https://github.com/maxi-blocks/maxi-blocks/files/12830916/Data.Company.txt)

Cool) better, because loading going block by block. not sure about minuses about it, but looks faster. Few questions:
1) settings in sidebar load step by step as well ( weird )
https://jam.dev/c/31f13270-f8de-487a-8abe-720ff167fd5d
2) https://jam.dev/c/77e380c3-3aa4-4d75-b54c-d716724d050d
loader for FSE looks bigger
3) reload page looks faster, but swapping between blocks, BP and settings looks much slower
https://jam.dev/c/28e2332a-a145-4c8b-bcdc-7af568fffe7f


with this page looks a bit better 
[Yoga 2.txt](https://github.com/maxi-blocks/maxi-blocks/files/12831005/Yoga.2.txt)

I guess, speed goes down if to add blog ( DC blocks )  inside page) I've added blog to Yoga 2 and it gets slower, like DataCompany Page

**Review by Olekrut — CHANGES_REQUESTED** [review]
Sorry, nothing has changed/  😢 
added video to point 1

**Comment by elzadj** [comment]
@Olekrut please remove the Map Icon 5 from the page and try again.

**Comment by Olekrut** [comment]
1) If compare by this video https://jam.dev/c/1048ea8c-678c-41d2-9034-334c0fc0eff1
we can see that reloading page is faster a bit ( difference 3 sec )
2) editor works slower
https://jam.dev/c/495f453d-59e6-4417-b8ee-b01887eb0b1c
3) styles for settings are loading after every tab opening ( looks not good )
https://jam.dev/c/9f7f714b-a035-4058-a58c-610b726faafc

**Comment by elzadj** [comment]
@Olekrut @kyrapieterse please try this one, or go to https://staging.maximaster.git.maxiblocks.com/ , I've installed it there

Also if you test speed on local, please do ```npm run build```, not ```npm start```

And ignore all failing tests, let's first figure out speed, then deal with tests.

**Comment by Olekrut** [comment]
1) bug with opacity everywhere. We have similar on master, but not the same
image

File: src/blocks/number-counter-maxi/edit.js [hunk 1]
```diff
@@ -3,17 +3,27 @@
  */
 import { useState, useEffect, useRef, createRef } from '@wordpress/element';
```
[the remark this answers is not in the document]
Comment (elzadj) [reply src/blocks/number-counter-maxi/edit.js:4]: @coderabbitai
it's used in 
```
constructor(props) {
		super(props);

		this.resizableObject = createRef();
	}
```

**Comment by elzadj** [comment]
@Olekrut 
> Margin, padding indicators feels slow
Could you pls elaborate? A video is ok

> Gutenberg`s BP feels slow ( our are better )
make new issue pls, it's not on optimization, we just made it work recently

**Comment by Olekrut** [comment]
https://jam.dev/c/4476e539-f159-49ff-b261-293c1b877f0c
about Margin and padding's indicators

**Comment by Olekrut** [comment]
https://jam.dev/c/0feb0b08-fbd9-4c49-b6b3-a5e572da585b
styles for settings are loading after every tab opening ( looks not good )

**Comment by Olekrut** [comment]
https://jam.dev/c/edfb0491-7de2-44eb-8c83-1ac3dc058798
duplicating looks weird

**Comment by elzadj** [comment]
> styles for settings are loading after every tab opening ( looks not good )
it will stay like this, speed is the priority

**Comment by elzadj** [comment]
@Olekrut @kyrapieterse updated on https://staging.3devdemo.maxiblocks.com/ , please test

**Comment by Olekrut** [comment]
https://jam.dev/c/a6bd91e0-09ce-428f-b540-2cfffb117bd7
Loader ( circle ) does not moving on Gutenberg`s BP and FSE

**Comment by Olekrut** [comment]
image
only for test site. Error. Load Accordion and try to open Pane

**Comment by Olekrut** [comment]
Appender higher than SC popover
image

**Comment by elzadj** [comment]
> Appender higher than SC popover, only simple editor, FSE is ok

@Olekrut please test now (local or https://staging.3devdemo.maxiblocks.com/)

**Review by Olekrut — APPROVED** [review]
🎉 approving, have some small requests that @elzadj  is aware of

**Pull request merged**